internal/httpx/httpx_test.go

```diff
@@ -93,6 +93,64 @@ func newAPIClient() *apiClient {
 	}
 }
 
+func TestJoinURLPath(t *testing.T) {
```

Comment:
This is very good! What would be excellent is to modify this test to be testing `newRequest`. Then you could set the `BaseURL` and the `resourcePath` like you're doing here and test whether the returned `http.Request`'s URL contains the desired path. I think this makes the test slightly more complex but give us one additional guarantee, i.e., that not only `joinURLPath` is correct but also `newRequest` uses the URL path.